From 89e30ccff1fe458810afb7851865318e38bcad5e Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 18 Sep 2014 14:20:14 -0400
Subject: [fs] xprtrdma: Make rpcrdma_ep_disconnect() return void

Message-id: <1411050057-12417-52-git-send-email-steved@redhat.com>
Patchwork-id: 96228
O-Subject: [RHEL7.1 PATCH 51/94] xprtrdma: Make rpcrdma_ep_disconnect() return void
Bugzilla: 1111708
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>

From: Chuck Lever <chuck.lever@oracle.com>

Clean up: The return code is used only for dprintk's that are
already redundant.

Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
Tested-by: Steve Wise <swise@opengridcomputing.com>
Tested-by: Shirley Ma <shirley.ma@oracle.com>
Tested-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Anna Schumaker <Anna.Schumaker@Netapp.com>
Upstream Id: 282191cb725db9a1aa80269e8369b06e9270a948
Bugzilla: 1111708
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/xprtrdma/transport.c b/net/sunrpc/xprtrdma/transport.c
index f6d280b..2faac49 100644
--- a/net/sunrpc/xprtrdma/transport.c
+++ b/net/sunrpc/xprtrdma/transport.c
@@ -414,7 +414,7 @@ xprt_rdma_close(struct rpc_xprt *xprt)
  if (r_xprt->rx_ep.rep_connected > 0)
   xprt->reestablish_timeout = 0;
  xprt_disconnect_done(xprt);
- (void) rpcrdma_ep_disconnect(&r_xprt->rx_ep, &r_xprt->rx_ia);
+ rpcrdma_ep_disconnect(&r_xprt->rx_ep, &r_xprt->rx_ia);
 }
 
 static void
diff --git a/net/sunrpc/xprtrdma/verbs.c b/net/sunrpc/xprtrdma/verbs.c
index f124f04..1208ab2 100644
--- a/net/sunrpc/xprtrdma/verbs.c
+++ b/net/sunrpc/xprtrdma/verbs.c
@@ -830,10 +830,7 @@ rpcrdma_ep_destroy(struct rpcrdma_ep *ep, struct rpcrdma_ia *ia)
  cancel_delayed_work_sync(&ep->rep_connect_worker);
 
  if (ia->ri_id->qp) {
-  rc = rpcrdma_ep_disconnect(ep, ia);
-  if (rc)
-   dprintk("RPC:       %s: rpcrdma_ep_disconnect"
-    " returned %i\n", __func__, rc);
+  rpcrdma_ep_disconnect(ep, ia);
   rdma_destroy_qp(ia->ri_id);
   ia->ri_id->qp = NULL;
  }
@@ -871,10 +868,8 @@ rpcrdma_ep_connect(struct rpcrdma_ep *ep, struct rpcrdma_ia *ia)
   struct rpcrdma_xprt *xprt;
 retry:
   dprintk("RPC:       %s: reconnecting...\n", __func__);
-  rc = rpcrdma_ep_disconnect(ep, ia);
-  if (rc && rc != -ENOTCONN)
-   dprintk("RPC:       %s: rpcrdma_ep_disconnect"
-    " status %i\n", __func__, rc);
+
+  rpcrdma_ep_disconnect(ep, ia);
   rpcrdma_flush_cqs(ep);
 
   if (ia->ri_memreg_strategy == RPCRDMA_FRMR)
@@ -984,7 +979,7 @@ out:
  * This call is not reentrant, and must not be made in parallel
  * on the same endpoint.
  */
-int
+void
 rpcrdma_ep_disconnect(struct rpcrdma_ep *ep, struct rpcrdma_ia *ia)
 {
  int rc;
@@ -1001,7 +996,6 @@ rpcrdma_ep_disconnect(struct rpcrdma_ep *ep, struct rpcrdma_ia *ia)
   dprintk("RPC:       %s: rdma_disconnect %i\n", __func__, rc);
   ep->rep_connected = rc;
  }
- return rc;
 }
 
 static int
diff --git a/net/sunrpc/xprtrdma/xprt_rdma.h b/net/sunrpc/xprtrdma/xprt_rdma.h
index 1ee6db3..c419498 100644
--- a/net/sunrpc/xprtrdma/xprt_rdma.h
+++ b/net/sunrpc/xprtrdma/xprt_rdma.h
@@ -341,7 +341,7 @@ int rpcrdma_ep_create(struct rpcrdma_ep *, struct rpcrdma_ia *,
     struct rpcrdma_create_data_internal *);
 void rpcrdma_ep_destroy(struct rpcrdma_ep *, struct rpcrdma_ia *);
 int rpcrdma_ep_connect(struct rpcrdma_ep *, struct rpcrdma_ia *);
-int rpcrdma_ep_disconnect(struct rpcrdma_ep *, struct rpcrdma_ia *);
+void rpcrdma_ep_disconnect(struct rpcrdma_ep *, struct rpcrdma_ia *);
 
 int rpcrdma_ep_post(struct rpcrdma_ia *, struct rpcrdma_ep *,
     struct rpcrdma_req *);
-- 
1.7.1